Make wxComboCtrl a wxCompositeWindow

a-wi opened the pull request:
wxComboCtrl consists of several controls (text entry, button, popup) and therefore should be implemented as a wxCompositeWindow to prevent problems with generating spurious events when e.g. focus is transferred between the sub-controls.

Closes https://trac.wxwidgets.org/ticket/18394

vadz commented on include/wx/combo.h:
```diff
@@ -475,6 +476,8 @@ class WXDLLIMPEXP_CORE wxComboCtrlBase : public wxControl,
     virtual bool SetForegroundColour(const wxColour& colour) wxOVERRIDE;
```
Shouldn't those 2 be removed/updated? `wxCompositeWindow` already propagates colours to all children, or do we want to avoid changing the colour of the button?

vadz reviewed (approved):
The changes here look fine, thanks! The hard part is to be sure if no other changes are required...

One nitpick: you need to use literal `Closes #18394` in the commit message instead of the full link for this to be picked up by wxTrac.

a-wi committed:
Make wxComboCtrl a wxCompositeWindow

wxComboCtrl consists of several controls (text entry, button, popup) and therefore should be implemented as a wxCompositeWindow to prevent problems with generating spurious events when e.g. focus is transferred between the sub-controls.

Closes #18394.
